common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -343,9 +343,23 @@ public MicrosoftStsTokenRequest createTokenRequest(@NonNull final MicrosoftStsAu
         tokenRequest.setClientId(request.getClientId());
         tokenRequest.setScope(request.getTokenScope());
         tokenRequest.setClaims(request.getClaims());
-        tokenRequest.setGrantType(TokenRequest.GrantTypes.AUTHORIZATION_CODE);
         setTokenRequestCorrelationId(tokenRequest);
 
+        // Existence of a device code inside of the response object implies Device Code Flow is being used
+        if (response.getDeviceCode() != null) {
+            tokenRequest.setGrantType(TokenRequest.GrantTypes.DEVICE_CODE);
+            tokenRequest.setDeviceCode(response.getDeviceCode());
+
+            // mTenantId from mAudience (being "common") is being added to the path in AzureActiveDirectoryAuthority
```

Comment:
Why was common the audience to begin with?

Reply:
The default authority in PCA defaults to having common as the tenant, I had to create a custom authority.